cirq/contrib/acquaintance/executor_test.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from itertools import combinations
+from typing import Sequence, Dict, Tuple
+
+import numpy as np
+import pytest
+
+import cirq
+from cirq.contrib.acquaintance.strategy import (
+    complete_acquaintance_strategy, AcquaintanceStrategy)
+from cirq.contrib.acquaintance.permutation import (
+        LinearPermutationGate)
+from cirq.contrib.acquaintance.executor import (
+        StrategyExecutor, GreedyExecutionStrategy)
+
+class ExampleGate(cirq.Gate, cirq.TextDiagrammable):
+    def __init__(self, wire_symbols: Sequence[str]) -> None:
+        self._wire_symbols = tuple(wire_symbols)
+
+    def text_diagram_info(self, args: cirq.TextDiagramInfoArgs
+                          ) -> cirq.TextDiagramInfo:
+        return cirq.TextDiagramInfo(
+                wire_symbols=self._wire_symbols)
+
+
+def test_executor_explicit():
+    n_qubits = 8
+    qubits = cirq.LineQubit.range(n_qubits)
+    circuit = complete_acquaintance_strategy(qubits, 2)
+
+    gates = {(i, j): ExampleGate([str(k) for k in ij])
```

Comment:
This is why you're getting inconsistent results. Dictionaries don't have a defined iteration order in older versions of python.

Reply:
I knew that the order was undefined and therefore could differ between machines and environments, but I'm very surprised it differs from *second to second* on the same machine seemingly randomly. Thanks for finding this.

Reply:
Some modern languages seed their hash functions differently every time you execute them, so that it's harder to perform denial of service attacks by causing hash collisions.